### PR TITLE
Use an action plugin instead of a list of tasks

### PR DESCRIPTION
**Pull request by cjeanner**
Using an action plugin should make it a bit faster and easier to call in
the future.

This PR has:
- [X] Appropriate testing (molecule, python unit tests)
- [X] Appropriate documentation (README in the role, main README is up-to-date)

### CODE REVIEW

**Review by raukadah — APPROVED** [review]
/lgtm

File: ci_framework/plugins/action/discover_latest_image.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        # Prepare matcher
+        matcher = re.compile(rf'{img_prefix}.*\.qcow2$')
+
+        # Start parsing
+        parsed = BeautifulSoup(page['content'], 'html.parser')
+        all_links = parsed.find_all('a')
+        extracted = []
+        for link in all_links:
+            if matcher.match(link.get('href')):
+                extracted.append(link.get('href'))
```
Comment (pablintino) [inline ci_framework/plugins/action/discover_latest_image.py:93]: Seems to be a good place to use list comprehension
```python
all_links = [a_elem.get('href') for a_elem in parsed.find_all('a')]
extracted =  [link for link in all_links if matcher.match(link)]
```

File: ci_framework/plugins/action/discover_latest_image.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        if BS4_IMPORT:
+            raise AnsibleError(missing_required_lib('beautifulsoup4'))
+
+        if 'image_prefix' not in module_args:
+            raise AnsibleError('"image_prefix" parameter is mandatory')
+
+        img_prefix = module_args.pop('image_prefix')
+
+        result = {
```
Comment (pablintino) [inline ci_framework/plugins/action/discover_latest_image.py:67]: Why not initializing these when used? At a first glance I expected this to be used in the following lines but it's used at near the end, when the whole result can be created in line 105

**Review by pablintino — COMMENTED** [review]
Only a few style comments

**Pull request merged**